File: 1-easy/problems/1572-matrix-diagonal-sum/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def diagonalSum(self, mat):
        """
        :type mat: List[List[int]]
        :rtype: int
        """
        return sum(mat[i][i]+mat[~i][i] for i in range(len(mat))) - (mat[len(mat)//2][len(mat)//2] if len(mat)%2 == 1 else 0)

File: 1-easy/problems/1572-matrix-diagonal-sum/solution.ts
function diagonalSum(mat: any): boolean | number | string | any {
  """
  :type mat: List[List[int]]
  :rtype: int
  """
  return sum(mat[i][i]+mat[~i][i] for i in Array.from({length: (mat}, (_, i) => i).length)) - (mat[(mat).length//2][(mat).length//2] if (mat).length%2 == 1 else 0);
}

export { diagonalSum };
